src/wp-includes/revision.php

```diff
@@ -277,35 +277,36 @@ function wp_save_post_revision( $post_id ) {
  * @return WP_Post|false The autosaved data or false on failure or when no autosave exists.
  */
 function wp_get_post_autosave( $post_id, $user_id = 0 ) {
-	global $wpdb;
-
-	$autosave_name = $post_id . '-autosave-v1';
-	$user_id_query = ( 0 !== $user_id ) ? "AND post_author = $user_id" : null;
-
-	// Construct the autosave query.
-	$autosave_query = "
-		SELECT *
-		FROM $wpdb->posts
-		WHERE post_parent = %d
-		AND post_type = 'revision'
-		AND post_status = 'inherit'
-		AND post_name   = %s " . $user_id_query . '
-		ORDER BY post_date DESC
-		LIMIT 1';
-
-	$autosave = $wpdb->get_results(
-		$wpdb->prepare(
-			$autosave_query,
-			$post_id,
-			$autosave_name
-		)
+	// Set up the query args for WP_Query.
+	$args = array(
+		'post_type'      => 'revision',
+		'post_status'    => 'inherit',
+		'post_parent'    => $post_id,
+		'post_name'      => $post_id . '-autosave-v1',  // Autosave name.
+		'posts_per_page' => 1,  // Only need the latest autosave.
+		'orderby'        => 'date', // Order by post date (latest first).
+		'order'          => 'DESC', // Descending order for latest.
+		'fields'         => 'ids', // We only need the post ID to fetch the post object.
```

Comment:
```suggestion
		'fields'         => 'ids', // We only need the post ID to fetch the post object.
		'no_found_rows'  => true,
```
Please add `no_found_rows`, as pagation is not required here.

Reply:
@spacedmonkey I have implemented the suggestions.